src/KeyringController.test.ts

```diff
@@ -1376,21 +1531,36 @@ describe('KeyringController', () => {
       );
     });
 
-    it('prepares a base UserOperation', async () => {
+    it('should throw if the keyring for the given address does not support signMessage', async () => {
+      const address = '0x5aC6d462f054690A373Fabf8cc28E161003aEB19';
+      stubKeyringClassWithAccount(BaseKeyringMock, address);
       const keyringController = await initializeKeyringController({
         password: PASSWORD,
       });
+      await keyringController.addNewKeyring(BaseKeyringMock.type);
+      const inputParams = {
+        from: address,
+        data: '0x879a053d4800c6354e76c7985a865d2922c82fb5b3f4577b2fe08b998954f2e0',
+        origin: 'https://metamask.github.io',
+      };
+
+      await expect(keyringController.signMessage(inputParams)).rejects.toThrow(
+        KeyringControllerError.UnsupportedSignMessage,
+      );
+    });
+  });
 
-      await keyringController.addNewKeyring('Keyring Mock With Init');
+  describe('prepareUserOperation', () => {
+    it('prepares a base UserOperation', async () => {
```

Comment:
Since the name of this test simply repeats the name of the method, are there any more details we can supply here? I'm noticing that the JSDocs for this method say "Convert a base transaction to a base UserOperation", so although it's a bit jargony, it paints a slightly better picture:

``` suggestion
    it('converts a base transaction to a base UserOperation', async () => {
```

or if we want to be a bit more literal:

``` suggestion
    it('calls the prepareUserOperation method on the keyring, returning its result', async () => {
```